docs_src/microservices/device/V2Migration.md

```diff
@@ -0,0 +1,213 @@
+SecretName = "redisdb"
+PublishTopicPrefix = 'edgex/events/device' # /<device-profile-name>/<device-name>/<source-name> will be added to this Publish Topic prefix
+  [MessageQueue.Optional]
+  # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+  # Client Identifiers
+  ClientId = "device-simple"
+  # Connection information
+  Qos = "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+  KeepAlive = "10" # Seconds (must be 2 or greater)
+  Retained = "false"
+  AutoReconnect = "true"
+  ConnectTimeout = "5" # Seconds
+  SkipCertVerify = "false" # Only used if Cert/Key file or Cert/Key PEMblock are specified
+```
+
+See the [Device Service MessageQueue](../../design/adr/013-Device-Service-Events-Message-Bus.md#device-services) section for details.
+
+### Code
+
+#### Dependencies
+You first need to update the `go.mod` file to specify `go 1.16` and the V2 versions of the Device SDK and any EdgeX go-mods directly used by your service. Note the extra `/v2` for the modules.
+
+!!! example "Example go.mod for V2"
+
+    ```go
+    module <your service>
+    
+    go 1.16
+    
+    require (
+    	github.com/edgexfoundry/device-sdk-go/v2 v2.0.0
+    	github.com/edgexfoundry/go-mod-core-contracts/v2 v2.0.0
+        ...
+    )
+    ```
+
+Once that is complete then the import statements for these dependencies must be updated to include the `/v2` in the path.
+
+!!! example "Example import statements for V2"
+
+    ```go
+    import (
+    	...
+        
+    	"github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+    	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+    )
+    ```
+
+#### CommandValue
+`CommandValue` is redesigned to be more simple and straightforward. A single `Value` with `interface{}` type is able to accommodate reading value of supported type.
+As a result, you might notice the original API to create CommandValue is no longer working.
+In V2 we refactor all those API functions to create CommandValue of different type to a generic function:
+
+!!! example "Create CommandValue with string Type"
+    ```go
+    cv, err := models.NewCommandValue(deviceResourceName, v2.ValueTypeString, "foobar")
+    if err != nil {
+        ...
+    }
+    cv.Origin = time.Now().Unixnano()
+    cv.Tags["foo"] = "bar"
+    ```
+
+The 3rd argument in the function must be able to cast into the Type defined in 2nd argument otherwise there will be error.
+See [Data formats](../../design/adr/device-service/0011-DeviceService-Rest-API.md#data-formats) for supported data type in EdgeX.
+
+Device Service also supports [Event Tagging](../general/index.md), the tags on the CommandValue will be copied to Event.
+
+### Device Profiles
+See [Device Profile Reference](profile/Ch-DeviceProfileRef.md) for details, SDK now allows both YAML and JSON format.
+
+#### Device Resource
+`properties` field is simplified in device resource.
```

Comment:
Add some statements:

- `units` becomes a single string field and it's optional.
- Float32 and Float64 are both only represented in eNotation.  Base64 encoding is removed, so there is no `floatEncoding` field anymore.